### ISSUE TITLE
Figures go missing from latex pdf if their files have the same base name and they use a  post transform

### ISSUE THREAD

**Issue by aaron-cooper** [sphinx-doc/sphinx#11110]
### Describe the bug

I'm trying to build a pdf from a document that contains some svg images. The images have identical names, but reside in different directories. I use `sphinxcontrib.rsvgconverter` from [sphinxcontrib-svg2pdfconverter](https://github.com/missinglinkelectronics/sphinxcontrib-svg2pdfconverter) to convert them from svg to pdf.

When multiple svg images with the same base name are passed to `rsvgconverter.convert`, the destination path for both images is identical, therefore all but the last image is discarded when it is overwritten by a subsequent transformation.

The result is that the final pdf will have the last converted svg repeated several times instead of each svg appearing when it is supposed to.

### How to Reproduce

[Here](https://github.com/aaron-cooper/sphinx-issue-demo) is a repo containing some code that reproduces the issue.

The file [foo_expected.pdf](https://github.com/aaron-cooper/sphinx-issue-demo/blob/main/foo_expected.pdf) (github doesn't show the full content of the pdf, remember to click "more pages" I guess...) shows what I would expect the output to be, and [foo_actual.pdf](https://github.com/aaron-cooper/sphinx-issue-demo/blob/main/foo_actual.pdf) shows what it actually is. Notice that the two identically named png files are shown correctly in foo_actual.pdf, but for the two svg's a red circle is shown twice, instead of a black circle and then a red circle.

### Environment Information

```text
Please paste all output below into the bug report template



Platform:              linux; (Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.35)
Python version:        3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0])
Python implementation: CPython
Sphinx version:        6.1.2
Docutils version:      0.19
Jinja2 version:        3.0.3
Pygments version:      2.13.0
```

**Comment by aaron-cooper** [comment]
I found a way to fix the issue, but I'm not savvy about this project so I have no idea if it would be a reasonable fix or not. Either way, [here is the commit](https://github.com/aaron-cooper/sphinx/commit/8d15b99b194aa42cef76ab8f7a18a5c3737e4fd0) from my fork, in case someone here finds that helpful.

**Comment by jfbu** [comment]
Good catch! I could reproduce this issue with
```
Running Sphinx v1.6.3
Sphinx version: 1.6.3
Python version: 3.7.5 (CPython)
Docutils version: 0.16 release
Pygments version: 2.14.0
Jinja2 version: 2.3
```
with `sphinxcontrig.svg2pdfconverter` at `1.0.0` and using Inkscape on my mac osx.

So this is a really long-standing issue... I will do some more testing.

**Comment by jfbu** [comment]
@aaron-cooper could you please make a PR with your patch? I used it in the

```diff
diff --git a/sphinx/transforms/post_transforms/images.py b/sphinx/transforms/post_transforms/images.py
index 1871a0f9c..410de82f0 100644
--- a/sphinx/transforms/post_transforms/images.py
+++ b/sphinx/transforms/post_transforms/images.py
@@ -236,7 +236,7 @@ class ImageConverter(BaseImageConverter):
         else:
             srcpath = node['candidates']['*']
 
-        filename = get_filename_for(srcpath, _to)
+        filename = get_filename_for(self.env.images[srcpath][1], _to)
         ensuredir(self.imagedir)
         destpath = os.path.join(self.imagedir, filename)
 
```
shape.

It appears to work fine. Sadly I am not much familiar with this myself.

Recently we have merged some changes with `sphinx.ext.imgmath` (see #10944, #10888 which was merged at 5.3.0 but did not get a CHANGES entry yet) but these things look very unrelated to sphinx/transforms/post_transforms/images.py, and so I see no problem there.

----

completely unrelated: in modifying a bit the test so that multiple `pdf`s end up in same page I encountered the 
```
pdfTeX warning: pdflatex (file ./img1.pdf): PDF inclusion: multiple pdfs with p
age group included in a single page
```
warning which is amply commented [as a tex.sx question](https://tex.stackexchange.com/questions/76273/multiple-pdfs-with-page-group-included-in-a-single-page-warning).  There is no such warning with `latex_engine` set to `'xelatex'` or `'lualatex'` although I do not know if the underlying problems may also arise there.  But this is more like something for [missinglinkelectronics/sphinxcontrib-svg2pdfconverter](https://github.com/missinglinkelectronics/sphinxcontrib-svg2pdfconverter) attention, and is not related in anyway to this issue.
